Prep for Spring, 2019

NicMcPhee opened the pull request:
There's some fairly substantial revisions to `README.md` and `LABTASKS.md` that address a number of outstanding issues.

The one really major change is a complete re-write of the tasks so that
everything now slices the cake. I have them creating an epic for each
feature, with (probably) two stories per epic (one for the server-side
API code, and one for the client-side support).

Things I tried to improve:

   * Specifying the JDK location in IntelliJ
   * Importing a project into IntelliJ
   * What it means to run the server
   * How to run JUnit tests
   * How to set up the Travis badge
   * Fix some broken links
   * Clarify the questions in `LABTASKS.md`
   * Link to some info on how create Epis in ZenHub

 I removed the material about creating a new Gradle task, as that
 didn't actually work and it certainly wasn't very clear.

 We changed from having them create an `ANSWERS.md` file in their repo
 to creating a GDoc with their answers and turning that in via Canvas.

Closes #14
Closes #27
Closes #29
Closes #31
Closes #33
Closes #39
Closes #40
Closes #41
Closes #42
Closes #48

kklamberty commented on ANSWERS.md:
```diff
@@ -0,0 +1,10 @@
+> You should put your answers to the questions in `LABTASKS.md`
```
Should it say, "There are questions in LABTASKS.md, and you should put the answers here in ANSWERS.md"? I first read this as you should put the answers in LABTASKS.md.

NicMcPhee replied:
So all that got completely rewritten when I switched from `ANSWERS.md` to GDoc, and I just deleted the `ANSWERS.md` file.

kklamberty commented on LABTASKS.md:
```diff
@@ -40,12 +40,12 @@ good to deliberately modify some of the tests and see what
 a `JsonObject` (like `UserController.getUser`). So at the moment we
 have solid testing of `User` and `Database`, but no direct testing 
-of `UserController`. We'll report back as we figure more out.
+of `UserController`.
```
I like this change. It does not promise anything new in the moment, which is a good thing.

kklamberty commented:
This is fine without any changes, but I found one thing that might need a little clarification. Looks great!

NicMcPhee commented:
@kklamberty – Can you look this over again? In particular, I *seriously* changed `LABTASKS.md` to now have things "slice the cake" more. I also changed things to use GDoc instead of `ANSWERS.md`. I've changed a lot in the past few hours, so having another pair of eyes look it over would be a Very Good Thing.

Thanks!

kklamberty commented:
I really like this new splitting of the lab tasks. Thanks for tackling that!
